tests/python/unittest/test_optimizer.py

```diff
@@ -357,6 +357,118 @@ def test_std_sparse_sgd():
                                           w_stype='row_sparse', g_stype='row_sparse')
 
 
+class PyNAG(PySGD):
+    def __init__(self, **kwargs):
+        super(PyNAG, self).__init__(**kwargs)
```

Comment:
Why inherit from PySGD in test??

Reply:
I think PySGD in test is appropriate for PyNAG to inherit. In SGD, _update_impl is responsible for the actual update (update and update_multi_precission both call this function). In PySGD, update is responsible for the actual update. So in PyNAG, overriding update function of PySGD is ok. This inheritance would make the impletation less verbose in test.